webtau-http-groovy/src/main/groovy/org/testingisdocumenting/webtau/http/datanode/GroovyDataNode.groovy

```diff
@@ -91,6 +92,11 @@ class GroovyDataNode implements DataNodeExpectations, DataNode {
         return node.elements().collect { new GroovyDataNode(it) }
     }
 
+    @Override
+    Collection<DataNode> children() {
```

Comment:
Should this be a stream so it cannot be modified by callers?

Reply:
It is protected for modification, not as implicit as `elements()`, but you can't change it. 
I would prefer to leave it as `Collection` to let folks, including us, do a good old `for(el in list) {}` loop

Reply:
What protects it from modification?  `StructuredDataNode.children()` returns `children.values()` and looking at the impl of `LinkedHashMap` for example I don't think it is protected from modification.  It seems to not allow insertions/updates but odly enough allows removals.  I tried this to be sure I'm not completely misunderstanding the code, this was my rather unsophisticated test:
```java
    public static void main(String[] args) {
        LinkedHashMap<String, String> map = new LinkedHashMap<>();
        map.put("foo", "bar");
        System.out.println(map);

        map.values().clear();
        System.out.println(map);
    }
```
Note the `values().clear()`, not direct `clear()` on the map.  This printed:
```
{foo=bar}
{}
```

Reply:
Oh, wow, thanks for checking - was under impression that it doesn't allow. Will wrap into Collections.unmodifiable

Reply:
I also assumed that it was essentially a view.